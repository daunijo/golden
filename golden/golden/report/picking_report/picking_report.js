// Copyright (c) 2016, RSS and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Picking Report"] = {
	"filters": [
		{
			"fieldname": "picking",
					"label": __("Picking"),
					"fieldtype": "Link",
					"options": "Picking"
		},
		{
			"fieldname": "sales_order",
					"label": __("Sales Order"),
					"fieldtype": "Link",
					"options": "Sales Order"
		},
		{
			"fieldname":"from_date",
			"label": __("From Date"),
			"fieldtype": "Date",
			"width": "80",
			"default": frappe.datetime.month_start()
		},
		{
			"fieldname":"to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			"width": "80",
			"default": frappe.datetime.month_end()
		},
	],
}
