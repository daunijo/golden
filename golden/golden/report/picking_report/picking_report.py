# Copyright (c) 2013, RSS and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe.utils import cint, flt, cstr
from frappe import _

def execute(filters=None):
	if not filters: filters = {}

	columns = get_columns()
	data = get_picking_report(filters)

	return columns, data

def get_columns():
	return [
	    #_("Status") + ":Data:80",
		_("Document") + ":Link/Picking:100",
		_("No.Sales Order")+":Link/Sales Order:100",
		_("Date")+":Date:80",
		_("Section")+":Data:150",
		_("CreatedDate") + ":Datetime:150",
		_("CreatedBy") + ":Data:150",
		_("ModifiedDate") + ":Datetime:150",
		_("ModifiedBy") + ":Data:150",
		_("Action")+":Data:100"
	]

def get_picking_report(filters):
	conditions = get_conditions(filters)
	return frappe.db.sql(

		"""select
				name,
				sales_order,
				transaction_date,
				section,
				creation,owner,
				modified,modified_by,
				if(docstatus = 1 && creation <= modified, "<span class = 'label label-success'>Auto</span>","<span class = 'label label-danger'>Manual</span>")

		   from
		   		`tabPicking`
		   where
		   		docstatus < 2 %s order by name desc,transaction_date desc
		""" % conditions, as_list=1)

def get_conditions(filters):
	conditions = ""
	if filters.get("picking"):
		conditions += "and name = '%s'" % filters["picking"]

	if filters.get("sales_order"):
		conditions += "and sales_order = '%s'" % filters["sales_order"]

	if filters.get("from_date"):
		conditions += "and transaction_date >= '%s'" % filters["from_date"]

	if filters.get("to_date"):
		conditions += "and transaction_date <= '%s'" % filters["to_date"]

	return conditions
