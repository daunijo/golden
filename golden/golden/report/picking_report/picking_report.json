{
 "add_total_row": 0, 
 "apply_user_permissions": 1, 
 "color": "#6e82b7", 
 "creation": "2018-02-14 10:26:31.680224", 
 "disabled": 0, 
 "docstatus": 0, 
 "doctype": "Report", 
 "icon": "octicon octicon-file", 
 "idx": 0, 
 "is_standard": "Yes", 
 "modified": "2018-02-14 10:26:31.680224", 
 "modified_by": "Administrator", 
 "module": "Golden", 
 "name": "picking report", 
 "owner": "Administrator", 
 "ref_doctype": "Picking", 
 "report_name": "picking report", 
 "report_type": "Script Report", 
 "roles": [
  {
   "role": "Administrator"
  }, 
  {
   "role": "Golden User"
  }
 ]
}